shotgun_api3/shotgun.py

```diff
@@ -1473,20 +1494,33 @@ def update_project_last_accessed(self, project, user=None):
         record = self._call_rpc("update_project_last_accessed_by_current_user", params)
         result = self._parse_records(record)[0]
 
-
-    def _get_session_token(self):
-        """Hack to authenticate in order to download protected content
-        like Attachments
+    def generate_session_token(self):
+        """
+        Generate a session token that can be used to authenticate the currently 
+        logged in user or script.
+        
+        :returns: String containing a session token
         """
-        if self.config.session_token:
-            return self.config.session_token
-
```

Comment:
The reason was that generate_session_token *generates* a new session token each time it is called. However, we may not need this as much now - it turns out the sessions don't expire after a certain time, they expire after a certain time of *inactivity*, making it (I think) pointless to keep creating new session tokens. As long as you keep using the token you have, you should be fine!

...which also means that we can probably return the cached token rather than making a new, and subsequently change it from generate_to_get() ! :)